interface/usergroup/facilities.php

```diff
@@ -2,62 +2,70 @@
+      "primary_business_entity" => trim(formData("primary_business_entity")),
+      "facility_npi" => trim(formData("facility_npi")),
+      "facility_code" => trim(formData("facility_id"))
+  );
+
+  $insert_id = $facilityService->insert($newFacility);
 }
 
 /*		Editing existing facility					*/
-if ($_POST["mode"] == "facility" && $_POST["newmode"] == "admin_facility")
+if (isset($_POST["mode"]) && $_POST["mode"] == "facility" && $_POST["newmode"] == "admin_facility")
 {
-	sqlStatement("update facility set
-		name='" . trim(formData('facility')) . "',
-		phone='" . trim(formData('phone')) . "',
-		fax='" . trim(formData('fax')) . "',
-		street='" . trim(formData('street')) . "',
-		city='" . trim(formData('city')) . "',
-		state='" . trim(formData('state')) . "',
-		postal_code='" . trim(formData('postal_code')) . "',
-		country_code='" . trim(formData('country_code')) . "',
-		federal_ein='" . trim(formData('federal_ein')) . "',
-		website='" . trim(formData('website')) . "',
-		email='" . trim(formData('email')) . "',
-		color='" . trim(formData('ncolor')) . "',
-		service_location='" . trim(formData('service_location')) . "',
-		billing_location='" . trim(formData('billing_location')) . "',
-		accepts_assignment='" . trim(formData('accepts_assignment')) . "',
-		pos_code='" . trim(formData('pos_code')) . "',
-		domain_identifier='" . trim(formData('domain_identifier')) . "',
-		facility_npi='" . trim(formData('facility_npi')) . "',
-		attn='" . trim(formData('attn')) . "' ,
-		primary_business_entity='" . trim(formData('primary_business_entity')) . "' ,
-		tax_id_type='" . trim(formData('tax_id_type')) . "' ,
-    facility_code = '" . trim(formData('facility_id')) . "'
-	where id='" . trim(formData('fid')) . "'" );
+  $newFacility = array(
+      "fid" => trim(formData("fid")),
+      "name" => trim(formData("facility")),
+      "phone" => trim(formData("phone")),
+      "fax" => trim(formData("fax")),
+      "street" => trim(formData("street")),
+      "city" => trim(formData("city")),
+      "state" => trim(formData("state")),
+      "postal_code" => trim(formData("postal_code")),
+      "country_code" => trim(formData("country_code")),
+      "federal_ein" => trim(formData("federal_ein")),
+      "website" => trim(formData("website")),
+      "email" => trim(formData("email")),
+      "color" => trim(formData("ncolor")),
+      "service_location" => trim(formData("service_location")),
+      "billing_location" => trim(formData("billing_location")),
+      "accepts_assignment" => trim(formData("accepts_assignment")),
+      "pos_code" => trim(formData("pos_code")),
+      "domain_identifier" => trim(formData("domain_identifier")),
+      "attn" => trim(formData("attn")),
+      "tax_id_type" =>  trim(formData("tax_id_type")),
+      "primary_business_entity" => trim(formData("primary_business_entity")),
+      "facility_npi" => trim(formData("facility_npi")),
+      "facility_code" => trim(formData("facility_id"))
```

Comment:
Better to do the sql injection in downstream class, so should remove all the formData calls (then will go to the Post parameter instead; see below for example on how to remove the formData):
``````
isset($_POST['facility']) ? $_POST['facility] : ''
...
``````